core/metadata.js

```diff
@@ -4,14 +4,15 @@ const _ = require('lodash')
 const { clone } = _
 const mime = require('mime')
 const deepDiff = require('deep-diff').diff
-const path = require('path')
-const { join } = path
+const nodepath = require('path')
 
 const logger = require('./logger')
 const { detectPathIssues, detectPathLengthIssue } = require('./path_restrictions')
 const { DIR_TYPE, FILE_TYPE } = require('./remote/constants')
 const { maxDate } = require('./timestamp')
 
+const fsutils = require('./utils/fs')
+const { extname, dirname, basename } = require('path')
```

Comment:
Out of curiosity, why not `const { ... } = nodepath`?
Do you prefer reusing `require`'s cache to make both require statements independent?

Reply:
I don't have any preferences, I use the same rule as 5 lines above.